### DISCUSSION TITLE
CRUD com React Js numa base de dados SQL (Postgres ou outro)

### DISCUSSION THREAD

**Discussion by asousajnri** [frontendbr/forum#1650] [Perguntas e Respostas]
Sou novo no React Js e estou tentando configurar um projeto para estudar a criação de um App React com alguma base de dado. Até agora não consegui muito, :( 

O máxima que consegui foi configurar o Apollo para consumir uma endoint e fazer alguns testes. 
https://github.com/asousajnri/pokemon-react

Ainda poderia me ajudar ou dá uma dica, caminho ou algo parecido para que eu consiga configurar um projeto React Js para interagir com um banco de dados e realizar um CRUD?

Tentei algumas coisas com API ORM (sequelize) e Graphql mais não dá forma que preciso.

Desde já agradeço!

Obs: não sei se a meu issue está dentro do padrão desejado, pois é o meu primeiro. Se tiver algo para melhorar nisso só da o feedback e corrigirei da melhor forma.

**Comment by valkervieira** [comment]
@asousajnri eu confesso que não entendi muito bem a sua dúvida. Você quer criar as rotas de uma API e interagir com o banco de dados através da aplicação de front? Ou só interagir com os endpoints de uma API já existente?

Se for o primeiro, seria melhor criar um servidor em [express](https://github.com/expressjs/express) ou [koa](https://github.com/koajs/koa) da vida e deixar o front só consumindo os endpoints (ou endpoint no caso de gql).

E quanto a afirmação do final, você usou o sequelize no front? O que exatamente você precisa?

**Comment by asousajnri** [comment]
Fala Valker, muito obrigado por responder, fiquei bem feliz.

Vou tentar exemplificar usando um cenário,

Criar quatro 4 páginas para as seguintes funções:
1. Uma página com um form de cadastro de pessoas;
2. Uma página para listar as pessoas cadastrados;
3. Uma página para editar um pessoas;
4. Uma para excluir pessoa(as);

Estou dividindo por página para ficar mais fácil de entender.

No caso utilizando React para front e algum outra API (Apollo, Sequelize
(ou outro) por exemplo) para consumir ou/e inserir os dados direto do banco
(Mysql ou Postgres etc), sem ser endpoint.

Não sei se a minha dúvida é coerente e se realmente é possível fazer o que
estou propondo.

De todo modo obrigado por ter me dado um retorno da dúvida.

Flw!

**Comment by kscastro** [comment]
@asousajnri Fala amigo,
Não entendi muito bem o ponto. Mas acho que voce vai precisar de um back-end para alimentar o banco de dados o Apollo e uma plataforma para voce conseguir se comunicar com o seu back-end feito em graphql. 

Qualquer duvida so chamar.

Abraços

**Comment by ninetails** [comment]
Tente _shiftar_ seu pensamento. O frontend ele apenas vai consumir APIs, vc precisa de um servidor backend para servir seus arquivos e expor essa API. O que o React vai fazer é consumir essa API.

Separe seu aprendizado em módulos (da mesma forma que também deveria separar as responsabilidades do seu app).
- Faça um backend que conversa com seu SQL e que consiga fazer o CRUD via API Rest
- _Desrestifique_ sua aplicação tentando fazê-la em GraphQL usando algum Playground
- Com a API pronta, faça o front se comunicar via GraphQL com seu servidor.

Tem um site generalista sobre GraphQL com os principais clients: https://www.howtographql.com/

Mas vc pode utilizar também a própria documentação da lib que vc optar (entre Apollo e Relay). Ambos creio que possuem tutoriais _fullstack_.

Creio que tb poderia ser possível começar direto com GraphQL pois alguns tutoriais básicos que vejo geralmente é com SQLite, daí portar para algum SQL, ORM ou outro serviço REST etc ficaria mais acessível.

**Comment by asousajnri** [comment]
Certo, obrigado galera!!! <3
